File: hw/gudino/u2/hw19 Inheritance/Java/ec.edu.espe_Farmsystem_jar_1.0-SNAPSHOT/src/main/java/ec/edu/espe/farmsymulator/model/Pig.java
package ec.edu.espe.farmsymulator.model;

import java.util.Date;

/**
 *
 * @author Bryan Gudino
 */
public class Pig extends FarmAnimal {

    public Pig(int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
    }

    @Override
    public String toString() {
        
        return "Pig{" + super.toString() + '}';
    }
    
}

File: hw/quina/u2/hw19Inheritance/FarmSystemJS/ec.edu.espe.FarmSystem.model/Pig.js
import { FarmAnimal } from './FarmAnimal.js';
import { Cage } from './Cage.js';

export class Pig extends FarmAnimal {
  constructor(id, breed, bornOn, gender, isAbleToReproduce, weight, cage /** @type {Cage} */) {
    super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
  }

  toString() {
    return `Pig(${super.toString()})`;
  }
}
